lib/bugsnag.rb

```diff
@@ -65,13 +65,27 @@ def notify(exception, auto_notify=false, &block)
         return
       end
 
+      # Store before_middleware severity reason for future reference
+      before_middleware_severity = report.severity
+      before_middleware_reason = report.severity_reason
+
       # Run users middleware
       configuration.middleware.run(report) do
         if report.ignore?
           configuration.debug("Not notifying #{report.exceptions.last[:errorClass]} due to ignore being signified in user provided middleware")
           return
         end
 
+        # Update severity reason if necessary
+        if report.severity != before_middleware_severity
+          report.severity_reason = {
+            :type => "userSpecifiedSeverity"
```

Comment:
The values for severity_reason should be defined as constants to avoid typos and to aggregate all possible values. Currently these values are scattered between different files.